src/App.tsx

```diff
@@ -1,14 +1,53 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoState } from './types/TodoState';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [inputQuery, setInputQuery] = useState('');
+  const [selectQuery, setSelectQuery] = useState<string>(TodoState.ALL);
+
+  const showSelectedTodo = (todoId: number) => {
+    const receivedTodo = todos.find(todo => todo.id === todoId) || null;
+
+    setSelectedTodo(receivedTodo);
+  };
+
+  useEffect(() => {
+    getTodos().then(todosFromServer => {
+      let filteredTodos = todosFromServer;
+      const preparedInputQuery = inputQuery.trim().toLowerCase();
+
+      if (preparedInputQuery) {
+        filteredTodos = filteredTodos.filter(todo =>
+          todo.title.toLowerCase().includes(preparedInputQuery),
+        );
+      }
+
+      if (selectQuery !== TodoState.ALL) {
+        if (selectQuery === TodoState.COMPLETED) {
+          filteredTodos = filteredTodos.filter(todo => todo.completed);
+        }
+
+        if (selectQuery === TodoState.ACTIVE) {
+          filteredTodos = filteredTodos.filter(todo => !todo.completed);
+        }
+      }
+
+      setTodos(filteredTodos);
+    });
+  });
```

Comment:
- You should load todos only on the component first render.
- The dependencies array is missing, so this `useEffect` doesn't make any sesnse
- You don't need to filter them inside this `useEffect`, you should filter when the query or filter type change
- Create a separate helper function to filter todos, move it to utils/helpers file and use switch case instead